### DISCUSSION TITLE
Additional CDR resource support

### DISCUSSION THREAD

**Discussion by loafoe** [philips-software/terraform-provider-hsdp#135] [Ideas]
Interesting discussion with @philips-software/hsdp-terraform  team on adding additional CDR resource types to the provider. Possible candidates:

- Location
- Practitioner
- Medication
- Device*

This would help in importing master data in CDR deployments. Opening this discussion to get feedback on possible schema formats and to look at practical applications

**Comment by jdelucaa** [comment]
Good one, exactly what I was looking for! 
We have a use case where we have a bunch of IAM users listed in the Terraform config for role assignments and that I would like to create as practitioners in the CDR as a alternative path while HSP does not have the ability to sync IAM users as practitioners. It would be very handy to be able to create Practitioners via Terraform since we would be able to create it from the list of IAM users we have there already. :)

**Reply by loafoe** [reply]
Nice! What fields would need to be there at a minimum for a practitioner resource for it to be usable to you?

**Reply by jdelucaa** [reply]
Hi @loafoe ,

As a starting point, it could support:
`meta`, `identifier` and `name`

Here is an example:

```
{
    "resourceType": "Practitioner",
    "meta": {
      "profile": [
        "StructureDefinition/Practitioner"
      ]
    },
    "identifier": [
      {
        "system": "test",
        "value": "888888888888"
      }
    ],
    "name": [
      {
        "text": "Person X",
        "given": [
          "Person"
        ],
        "family": "X"
      }
    ]
}
```

**Reply by loafoe** [reply]
@jdelucaa `hsdp_cdr_practitioner` is  available in `v0.35.0` and newer: https://registry.terraform.io/providers/philips-software/hsdp/latest/docs/resources/cdr_practitioner
